src/nvim/api/fold.c

```diff
@@ -0,0 +1,42 @@
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "nvim/api/fold.h"
+#include "nvim/api/private/validate.h"
+#include "nvim/fold.h"
+#include "nvim/pos_defs.h"
+
+#ifdef INCLUDE_GENERATED_DECLARATIONS
+# include "api/fold.c.generated.h"  // IWYU pragma: keep
+#endif
+
+/// Gets the start and end lines for the closed fold on the given line. If the line is not within a
+/// closed fold, an empty table will be returned.
+///
+/// @param window   |window-ID|, or 0 for current window
+/// @param lnum   The line number to check (0-indexed)
+/// @param[out] err Error details, if any
+/// @return {start, stop}
+ArrayOf(Integer, 2) nvim__fold_info_for_line(Window window, Integer lnum, Arena *arena, Error *err)
```

Comment:
Even if this only returns start/end for now, let's call it `nvim__fold_info` and return a Dict so we can expand it later?
```suggestion
Dict(fold_info) nvim__fold_info(Window window, Integer lnum, Arena *arena, Error *err)
```